### PR TITLE
8294821: Class load improvement for AES crypto engine

### PR DESCRIPTION
**Pull request by XueleiFan**
Hi,

May I have the code clean up reviewed?
 
There is a lot of computation in AESCrypt class load, which could be avoid by using the computation result directly. The computation takes 6.971875 milliseconds in a MacOS M1 laptop. Although it is a one-time computation, but removing the computation could speed up java/jvm loading time by 6.971875 milliseconds without extra memory cost. 

Actually, because the computation result are used directly, the methods that supports the computation are no longer needed and the related static arrays are no longer needed.  This is the extra benefits to use the computation result directly, rather than computing on class initialization.

This patch does not change the logic except removing the pre-computation code and initializing the tables with the pre-computed result explicitly.  The existing regression and inter-op tests should be sufficient to ensure that the tables are correctly copied from the dumping of the old computation code results.

Except that, I also cleaned up some code warnings from the IDE I used.

Thanks,
Xuelei

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294821](https://bugs.openjdk.org/browse/JDK-8294821): Class load improvement for AES crypto engine

### Reviewers
 * [Valerie Peng](https://openjdk.org/census#valeriep) (@valeriepeng - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10568/head:pull/10568` \
`$ git checkout pull/10568`

Update a local copy of the PR: \
`$ git checkout pull/10568` \
`$ git pull https://git.openjdk.org/jdk pull/10568/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10568`

View PR using the GUI difftool: \
`$ git pr show -t 10568`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10568.diff">https://git.openjdk.org/jdk/pull/10568.diff</a>

</details>

### CODE REVIEW

**Comment by dfuch** [comment]
I wonder if it would be worthwhile to have a static method that could be called in an assert (or in a test using --patch-module) to verify that the various statically initialized arrays are the same than those that would have been computed by the old code?

**Comment by XueleiFan** [comment]
> I wonder if it would be worthwhile to have a static method that could be called in an assert (or in a test using --patch-module) to verify that the various statically initialized arrays are the same than those that would have been computed by the old code?

I did think about the same idea, by putting the code into test codebase or generate the tables with makefile configuration.   Actually, I had the table generation code in my local workspace.  However, I give it up at last because such kind of code is just used one time and may not worthy the resources to maintain it or the power to run it while building/testing.  None of the tables could be wrong, otherwise the algorithm will not work.  The regression test and inter-op testing may be sufficient to approve that the tables are good.

But please let me know, if you have a strong preference to have it, I will try to integrate the table generation code.

**Comment by dfuch** [comment]
> But please let me know, if you have a strong preference to have it, I will try to integrate the table generation code.

Your explanation sounds reasonable to me - but I will leave it to security-libs reviewers and maintainers to make the call.

**Comment by mcpowers** [comment]
Changing one pre-computed byte should be sufficient to blow up all AES tests.
Add a comment saying where your pre-computed values come from, e.g. Tables from FIPS 197.

File: src/java.base/share/classes/com/sun/crypto/provider/AESCrypt.java
Comment (XueleiFan) [reply src/java.base/share/classes/com/sun/crypto/provider/AESCrypt.java:1065]: Good point!

**Review by valeriepeng — APPROVED** [review]
Look good. Thanks for the patch~

**Comment by valeriepeng** [comment]
BTW,  probably should add a noreg-perf to the bug.

**Comment by XueleiFan** [comment]
Anyone can help run Mach5 testing in case I missed something?

**Comment by valeriepeng** [comment]
> Anyone can help run Mach5 testing in case I missed something?

Sure, I can help with that.

**Comment by valeriepeng** [comment]
Mach5 test looks clean. There is one unexpected test failure but it seems unrelated (https://mach5.us.oracle.com:10060/api/v1/results/vpeng-jdkOh-20221010-1957-37280778-open_test_lib-test-linux-x64-122-1665432715-16/log) . So, should be fine.

**Pull request closed without merging**
